client/src/containers/SubmitCode.js

```diff
@@ -64,5 +95,25 @@ class SubmitCode extends Component {
 	}
 }
 
-export default connect(
-)(SubmitCode);
+SubmitCode.contextTypes = { history: React.PropTypes.object.isRequred }
+
+SubmitCode.propTypes = {
+    user: PropTypes.object,
+    code: PropTypes.object
+}
+
+function mapStateToProps(state) {
+    var user = state.user.user; 
+    var code = {
```

Comment:
Same comment as before, maybe have code be in the local state. Unless you've seen this done somewhere else because I'm not an expert on redux style. Definitely comment if you have.